app/controllers/authenticate_controller.rb

```diff
@@ -70,7 +70,8 @@ def authenticate(input = authenticator_input)
       authenticators: installed_authenticators,
       enabled_authenticators: Authentication::InstalledAuthenticators.enabled_authenticators_str(ENV)
     )
-    render json: authn_token
+    content_type = input[:encoded_token] ? :plain : :json
```

Comment:
I agree with the guideline about the endpoint. But the encoded response is not a json so a separate endpoint sounds reasonable.

Reply:
@liavyona can you please investigate on a best practice for this?

Reply:
endpoint should stick with its content-type. btw, requests also ( requests and response type should be the same). But this is out of this PR's scope

Reply:
why is it out of the scope of this PR? this PR adds a new type of response and should be handled in the proper way. if the best practice is to have just one content type then the correct implementation of this PR is to add a new endpoint

Reply:
I talked about one content-type to requests and response throughout the entire project

Reply:
so i will ask again - is it considered a bad practice to have one endpoint return different types of response depending on the input?

Reply:
Yes

Reply:
well in that case, let's create a new endpoint and not introduce a bad practice. Sorry that we got to this point after you already implemented this but it's better to catch it now rather than when it's in master.

let me know if you need any assistance in implementing a new endpoint (e.g `.../authenticate/encoded`)

Reply:
We can open it for discussion (with architact?) since it's best practice, and not an error to implement it as such it can go either way - so we can hold on with implementing another endpoint, @liavyona do we have another endpoint that act the same?

Reply:
> well in that case, let's create a new endpoint and not introduce a bad practice. Sorry that we got to this point after you already implemented this but it's better to catch it now rather than when it's in master.
> 
> let me know if you need any assistance in implementing a new endpoint (e.g `.../authenticate/encoded`)

It's Ok. Just adding it to route.rb . Thanks.

Reply:
> We can open it for discussion (with architact?) since it's best practice, and not an error to implement it as such it can go either way - so we can hold on with implementing another endpoint, @liavyona do we have another endpoint that act the same?

I'll look for this kind of behaviour. But I'm pretty sure there are no such cases (from search "render" in the project)